2996_reflection/reflection.md

```diff
@@ -4025,6 +4013,21 @@ Adjust paragraph 3 of [temp.arg.general] to not apply to splice template argumen
 
 [3]{.pnum} [A `$template-argument$` of the form `$splice-specifier$` is interpreted as a `$splice-template-argument$`.]{.addu} In a `$template-argument$` [that is not a `$splice-template-argument$`]{.addu}, an ambiguity between a `$type-id$` and an expression is resolved to a `$type-id$`, regardless of the form of the corresponding `$template-parameter$`.
 
+::: example2
+```diff
+template<class T> void f();
+template<int I> void f();
+
+void g() {
+  f<int()>();       // int() is a type-id: call the first f()
+
++ constexpr int x = 42;
++ f<[:^int:]>();    // splice-template-argument: calls the first f()
++ f<[:^x:]>();      // splice-template-argument: calls the second f()
+}
```

Comment:
I like adding spaces in everything so that things just line up better (also so the `+`s don't compete with the `}` vertically, makes it stand out a little more):

```suggestion
  template<class T> void f();
  template<int I> void f();

  void g() {
    f<int()>();       // int() is a type-id: call the first f()

+   constexpr int x = 42;
+   f<[:^int:]>();    // splice-template-argument: calls the first f()
+   f<[:^x:]>();      // splice-template-argument: calls the second f()
  }
```